webapp/src/components/Notification/NotificationList.tsx

```diff
@@ -0,0 +1,222 @@
+  }
+`;
+
+const NotificationItem = styled.div`
+  display: flex;
+  padding: 16px;
+  width: 100%;
+
+  &:hover {
+    background: rgba(255, 255, 255, 0.04);
+  }
+`;
+
+const NotificationItemIcon = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  height: 40px;
+  width: 40px;
+`;
+
+const NotificationItemInfo = styled.div`
+  display: flex;
+  flex-direction: column;
+  width: 249px;
+`;
+
+const NotificationItemVaultPill = styled.div<{ color: string }>`
+  display: flex;
+  padding: 4px 8px;
+  background: ${(props) => props.color}29;
+  border-radius: 100px;
+`;
+
+interface NotificationListProps {
+  onClose: () => void;
+  filters: NotificationType[];
+}
+
+const NotificationList: React.FC<NotificationListProps> = ({
+  onClose,
+  filters,
+}) => {
+  const notifications = useNotifications();
+  const history = useHistory();
+
+  const filteredNotifications = useMemo(
+    () =>
+      notifications.filter((notification) =>
+        filters.includes(notification.type)
+      ),
+    [filters, notifications]
+  );
+
+  const renderNotificationInfo = useCallback((notification: Notification) => {
+    const asset = getAssets(notification.vault);
+    const decimals = getAssetDecimals(asset);
+    const color = getVaultColor(notification.vault);
+
+    const badge = (
+      <NotificationItemVaultPill color={color} className="ml-2">
+        <Subtitle fontSize={10} lineHeight={12} color={color}>
+          {productCopies[notification.vault].title}
+        </Subtitle>
+      </NotificationItemVaultPill>
+    );
+
+    switch (notification.type) {
+      case "optionMinting":
+        return (
+          <>
+            <div className="d-flex align-items-center">
+              <Title fontSize={14} lineHeight={24}>
+                MINTED OPTIONS
+              </Title>
+              {badge}
+            </div>
+            <PrimaryText
+              fontSize={12}
+              lineHeight={16}
+              color={colors.text}
+              className="mt-1"
+            >
+              The vault minted{" "}
+              {formatBigNumber(notification.depositAmount, decimals)} options at
+              a strike price of $
+              {formatOption(notification.strikePrice).toLocaleString()}
+            </PrimaryText>
+          </>
+        );
+      case "optionSale":
+        return (
+          <>
+            <div className="d-flex align-items-center">
+              <Title fontSize={14} lineHeight={24}>
+                SOLD OPTIONS
+              </Title>
+              {badge}
+            </div>
+            <PrimaryText
+              fontSize={12}
+              lineHeight={16}
+              color={colors.text}
+              className="mt-1"
+            >
+              The vault sold{" "}
+              {formatOption(notification.sellAmount).toLocaleString()} options
+              for {formatBigNumber(notification.premium, decimals)}{" "}
+              {getAssetDisplay(asset)}
```

Comment:
We can refactor this quite a bit, here's my suggestion:

```js
let notificationBody

switch (notification.type) {
    case "optionMinting":
        notificationBody = <>
            ...
        </>
        break
    case "optionSale":
        notificationBody = <>
            ...
        </>
        break
    case "withdrawalReady":
        notificationBody = <>
            ...
        </>
        break
    default:
        break;
}


return (
    <>
    <div className="d-flex align-items-center">
        <Title fontSize={14} lineHeight={24}>
        MINTED OPTIONS
        </Title>
        {badge}
    </div>
    <PrimaryText
        fontSize={12}
        lineHeight={16}
        color={colors.text}
        className="mt-1"
    >
        {notificationBody}
    </PrimaryText>
    </>
);
```

Reply:
Over here, I actually think the hook approach would result in better performance as the page rerender frequently, and as long as the hook dependencies does not change in value, there will not be rerendering for the notification body happen. 

Without encapsulating it inside a `useMemo` or `useCallback`, they will be rerender every single time the page are rerender